.github/workflows/build-wheels-publish.yml

```diff
@@ -0,0 +1,303 @@
+          # macos - arm64
+          - { os: macos-latest, arch: arm64, cibw_build: 'cp38', cibw_skip: '' }
+          # windows - x86_64
+          - { os: windows-latest, arch: AMD64, cibw_build: 'cp38', cibw_skip: '' }
+          # windows - amd64
+          - { os: windows-latest, arch: x86, cibw_build: 'cp38', cibw_skip: '' }
+          # windows - arm64
+          - { os: windows-latest, arch: ARM64, cibw_build: 'cp39', cibw_skip: '' }
+
+    steps:
+      - uses: actions/checkout@v4
+
+      # https://cibuildwheel.pypa.io/en/stable/faq/#macos-building-cpython-38-wheels-on-arm64
+      - uses: actions/setup-python@v5
+        if: runner.os == 'macOS' && runner.arch == 'ARM64'
+        with:
+          python-version: 3.8
+
+      - name: '🛠️ Win MSVC 32 dev cmd setup'
+        if: runner.os == 'Windows' && matrix.arch == 'x86'
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: x86
+
+      - name: '🛠️ Win MSVC 64 dev cmd setup'
+        if: runner.os == 'Windows' && matrix.arch == 'AMD64'
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: x64
+
+      - name: '🛠️ Win MSVC ARM64 dev cmd setup'
+        if: runner.os == 'Windows' && matrix.arch == 'ARM64'
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: amd64_arm64
+
+      - name: '🛠️ Set up QEMU'
+        if: runner.os == 'Linux'
+        uses: docker/setup-qemu-action@v3
+
+      - name: '🚧 cibuildwheel run'
+        uses: pypa/cibuildwheel@v2.21.3
+        env:
+          CIBW_BUILD_FRONTEND: build
+          CIBW_BUILD: ${{ matrix.cibw_build }}*
+          CIBW_SKIP: ${{ matrix.cibw_skip }}
+          CIBW_ARCHS: ${{ matrix.arch }}
+          CIBW_ENVIRONMENT: DEBUG=${{ env.CAPSTONE_DEBUG }}
+          CIBW_ENVIRONMENT_PASS_LINUX: DEBUG
+          # https://cibuildwheel.pypa.io/en/stable/faq/#windows-arm64
+          # https://github.com/pypa/cibuildwheel/pull/1169
+          CIBW_TEST_SKIP: "*-win_arm64 cp38-macosx_*:arm64"
+          CIBW_TEST_COMMAND: >
+            python -m pip install {package}/cstest_py &&
+            python {package}/tests/test_all.py &&
+            cd {project}/suite/cstest/test/ &&
+            python integration_tests.py cstest_py &&
+            cd {project} &&
+            cstest_py {project}/tests/MC/ &&
+            cstest_py {project}/tests/details/ &&
+            cstest_py {project}/tests/issues/ &&
+            cstest_py {project}/tests/features/
+          CIBW_TEST_COMMAND_WINDOWS: >
+            python -m pip install {package}/cstest_py &&
+            python {package}/tests/test_all.py &&
+            cd /d {project}/suite/cstest/test/ &&
+            python integration_tests.py cstest_py &&
+            cd /d {project} &&
+            cstest_py {project}/tests/MC/ &&
+            cstest_py {project}/tests/details/ &&
+            cstest_py {project}/tests/issues/ &&
+            cstest_py {project}/tests/features/
```

Comment:
not sure worth it having them into separate scripts (per platform). Meanwhile I have pushed some fixes that made simplify the cibuildwheel tests command. The scripts would just add a print and after all you would need to walk the logs anyway, and they already contain a summary of what tests have failed.

Reply:
Though about it. But it is worth having tem in an extra script, because it can be run locally. Otherwise, if people do changes to these lines, they have to push the file or run `act` or similar to test it. Both is rather time and resource consuming.
SO a script is better for testing.

You can move the `python -m pip install {package}/cstest_py` into a separated step. Because they are the same for Windows/Linux.

Reply:
Added `run_tests.py` script under `suite` folder and adapted workflow accordingly